src/async-wrap.cc

```diff
@@ -616,7 +617,22 @@ AsyncWrap::AsyncWrap(Environment* env,
   persistent().SetWrapperClassId(NODE_ASYNC_ID_OFFSET + provider);
 
   // Use AsyncReset() call to execute the init() callbacks.
-  AsyncReset(silent);
+  AsyncReset(eid);
+}
+
+
+AsyncWrap::AsyncWrap(Environment* env,
+                     Local<Object> object,
+                     bool silent)
+    : BaseObject(env, object),
+      provider_type_(PROVIDER_PROMISE) {
```

Comment:
> I can't follow that argument at all. To me, using default arguments is only an issue if there become too many of them, I don't see that happening here.

If there's a parameter that will only ever be used by one other class, in the same file, I don't see a good reason that it should pollute the general constructor. Even if it has a default.

> Besides, your solution is just a different kind of default argument.

Don't follow what you're saying.

> It is not clear from the calling code.

I'm really not worried about that since it's only called once from the same file.